Sources/Sentry/SentryProfiler.mm

```diff
@@ -170,28 +177,42 @@
 
 #    if SENTRY_HAS_UIKIT
 NSArray *
-processFrameRenders(SentryFrameInfoTimeSeries *frameInfo, uint64_t start, uint64_t duration)
+processFrameRenders(
+    SentryFrameInfoTimeSeries *frameInfo, uint64_t profileStart, uint64_t profileDuration)
 {
     auto relativeFrameInfo = [NSMutableArray array];
     [frameInfo enumerateObjectsUsingBlock:^(
         NSDictionary<NSString *, NSNumber *> *_Nonnull obj, NSUInteger idx, BOOL *_Nonnull stop) {
-        const auto begin = timeIntervalToNanoseconds(obj[@"start_timestamp"].doubleValue);
-        if (begin < start) {
+        const auto frameRenderStart
+            = timeIntervalToNanoseconds(obj[@"start_timestamp"].doubleValue);
+
+#        if defined(TEST) || defined(TESTCI)
+        // we don't currently validate the timestamps in tests, and the mock doesn't provide
+        // realistic ones, so they'd fail the checks below. just write them directly into the data
+        // structure so we can count *how many* were recorded
+        [relativeFrameInfo addObject:@{
+            @"elapsed_since_start_ns" : @(frameRenderStart),
+            @"value" : @(frameRenderStart),
+        }];
+        return;
```

Comment:
Hmm, I thought about it but wasn't quite sure how to mock the values so that the original logic won't wind up crashing with negative durations, which is what the early returns in the logic are for. This workaround essentially allows us to ensure that the frame tracker did in fact record frame render info and pass it to the profiler. As I mentioned in https://github.com/getsentry/sentry-cocoa/pull/2493#issuecomment-1380915260, I will write more tests to ensure the slicing logic in here and added in that PR are tested for correctness. Are you OK kicking that can down the road a bit more?

Reply:
> Are you OK kicking that can down the road a bit more?

Obviously, otherwise, I wouldn't have approved your PR.